test/core/messages.integration.test.ts

```diff
@@ -441,4 +443,17 @@ describe('messages integration', () => {
     // Calling off again should be a no-op
     off();
   });
+
+  it<TestContext>('handles the room being released before getPreviousMessages is called', async (context) => {
+    const chat = context.chat;
+    const roomId = randomRoomId();
+    const room = chat.rooms.get(roomId, RoomOptionsDefaults);
+
+    // Create a subscription to messages
+    room.messages.subscribe(() => {});
+
+    // Now release the room
+    // We should not have any unhanded promise rejections
+    await chat.rooms.release(roomId);
```

Comment:
Is this test working correctly if there are unhandled promise rejections? Those errors get thrown outside of the normal execution, so they don't typically make anything crash.

Not sure how to listen to them exactly - maybe listening to https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event?

Reply:
The test will still "pass" - unfortunately we can't get around that. But the actual test process will exit with code `1` and will therefore fail CI.

I've added a comment to this effect.